v-next/hardhat/test/internal/builtin-plugins/console/task-action.ts

```diff
@@ -0,0 +1,182 @@
+      const replServer = await consoleAction(
+        {
+          commands: ['await import("./scripts/non-existent.js");', ".exit"],
+          history: "",
+          noCompile: false,
+          options,
+        },
+        hre,
+      );
+      ensureError(replServer.lastError);
+    });
+
+    it("should run a script inside the console successfully", async function () {
+      const replServer = await consoleAction(
+        {
+          commands: [".help", 'await import("./scripts/success.js");', ".exit"],
+          history: "",
+          noCompile: false,
+          options,
+        },
+        hre,
+      );
+      assert.equal(replServer.lastError, undefined);
+    });
+
+    it("should throw inside the console if the script throws", async function () {
+      const replServer = await consoleAction(
+        {
+          commands: ['await import("./scripts/throws.js");', ".exit"],
+          history: "",
+          noCompile: false,
+          options,
+        },
+        hre,
+      );
+      ensureError(replServer.lastError);
+    });
+  });
+
+  describe("typescript", function () {
+    useFixtureProject("run-ts-script");
+
+    it("should throw inside the console if script does not exist", async function () {
+      const replServer = await consoleAction(
+        {
+          commands: ['await import("./scripts/non-existent.ts");', ".exit"],
+          history: "",
+          noCompile: false,
+          options,
+        },
+        hre,
+      );
+      ensureError(replServer.lastError);
+    });
+
+    it("should run a script inside the console successfully", async function () {
+      const replServer = await consoleAction(
+        {
+          commands: ['await import("./scripts/success.ts");', ".exit"],
+          history: "",
+          noCompile: false,
+          options,
+        },
+        hre,
+      );
+      assert.equal(replServer.lastError, undefined);
+    });
+
+    it("should throw inside the console if the script throws", async function () {
+      const replServer = await consoleAction(
+        {
+          commands: ['await import("./scripts/throws.ts");', ".exit"],
+          history: "",
+          noCompile: false,
+          options,
+        },
+        hre,
+      );
+      ensureError(replServer.lastError);
+    });
+  });
+
+  describe("context", function () {
+    it("should expose the Hardhat Runtime Environment", async function () {
+      const replServer = await consoleAction(
+        {
+          commands: ["console.log(hre);", ".exit"],
+          history: "",
+          noCompile: false,
+          options,
+        },
+        hre,
+      );
+      assert.equal(replServer.lastError, undefined);
+    });
+  });
+
+  describe("history", function () {
+    let cacheDir: string;
+    let history: string;
+
+    beforeEach(async function () {
+      cacheDir = await fsPromises.mkdtemp(
+        path.resolve(os.tmpdir(), "console-action-test-"),
+      );
```

Comment:
Yes, let's add it to the utils instead. Our intention is to avoid using `fs` anywhere except that one module. We'll soon have a linter rule for that.

But let's do it in a follow up PR. Can you do it/create an issue about it?

Reply:
@schaable tagging you just for visibility.

Reply:
I created the issue here - https://github.com/NomicFoundation/hardhat/issues/5601